### PR TITLE
Exclude carriage return from search results on Windows

### PR DESCRIPTION
**Pull request by Kamholtz**
Fix for #87 which removes the `\r` from the end of search results on Windows.

### CODE REVIEW

File: lua/grug-far/rg/parseResults.lua [hunk 1]
```diff
@@ -88,6 +89,26 @@ local function getStats(tokens)
...
+  if not is_win then
+    return line
+  end
+
+  local last_char = string.sub(line, -1)
+  if last_char ~= '\r' then
+    return line
+  end
+
+  if string.len(line) <= 1 then
```
Comment (MagicDuck) [inline lua/grug-far/rg/parseResults.lua:105]: why is this if-block necessary? `string.sub('', 1, -2)` is `''` and we don't care about a lone carriage return, right?

**Comment by MagicDuck** [comment]
Great! Thanks for the contrib! I don't have windows myself so it's much appreciated when somebody fixes those! 😄 

Just a small comment.

Oh also, don't want to presume on your help, but if you can, could you also see if we should remove `utils.eol` (and just replace with `\n` maybe) ?:
```
https://github.com/search?q=repo%3AMagicDuck%2Fgrug-far.nvim+utils.eol&type=code
```
That was my poor attempt at windows support with no windows machine on which to test, haha

File: lua/grug-far/rg/parseResults.lua (hunk 1 above)
Comment (Kamholtz) [reply lua/grug-far/rg/parseResults.lua]: Good point, I've added a change to remove that if-block.

**Comment by Kamholtz** [comment]
> Great! Thanks for the contrib! I don't have windows myself so it's much appreciated when somebody fixes those! 😄
> 
> Just a small comment.
> 
> Oh also, don't want to presume on your help, but if you can, could you also see if we should remove `utils.eol` (and just replace with `\n` maybe) ?:
> 
> https://github.com/search?q=repo%3AMagicDuck%2Fgrug-far.nvim+utils.eol&type=code
> 
> This is for sync action. That was my poor attempt at windows support with no windows machine on which to test, haha I can't tell if it actually helps in this situation or not. Windows eol is confusing...

I'm happy to help, thank you for taking the time review my PR!  I should be able to take look at removing `utils.eol` and testing on Windows in a day or two :)

**Review by MagicDuck — APPROVED** [review]

**Comment by MagicDuck** [comment]
> I'm happy to help, thank you for taking the time review my PR! I should be able to take look at removing utils.eol and testing on Windows in a day or two :)

Great! :happy dance: 😄 
I'll merge this one since it looks good and could help other people and we can have the `utils.eol` stuff be a separate PR as needed!

**Comment by MagicDuck** [comment]
oops, pressed wrong button

**Pull request merged**